File: brace_checker.js
const fs = require('fs');

const content = fs.readFileSync('src/components/dashboard/DashboardGuide.tsx', 'utf-8');
const lines = content.split('\n');

let braceCount = 0;
let bracketCount = 0;
let parenCount = 0;

for (let i = 0; i < lines.length; i++) {
    const line = lines[i];
    const bOpen = (line.match(/{/g) || []).length;
    const bClose = (line.match(/}/g) || []).length;
    const sqOpen = (line.match(/\[/g) || []).length;
    const sqClose = (line.match(/\]/g) || []).length;
    const pOpen = (line.match(/\(/g) || []).length;
    const pClose = (line.match(/\)/g) || []).length;

    braceCount += bOpen - bClose;
    bracketCount += sqOpen - sqClose;
    parenCount += pOpen - pClose;

    if (braceCount < 0) {
        console.log(`Brace Underflow at line ${i + 1}: ${line.trim()}`);
        break;
    }
    if (bracketCount < 0) {
        console.log(`Bracket Underflow at line ${i + 1}: ${line.trim()}`);
        break;
    }
}

console.log(`Final Counts - Brace: ${braceCount}, Bracket: ${bracketCount}, Paren: ${parenCount}`);
if (braceCount === 0 && bracketCount === 0 && parenCount === 0) {
    console.log("Balanced!");
} else {
    console.log("Imbalanced!");
}

File: brace_checker.py
import sys

with open('src/components/dashboard/DashboardGuide.tsx', 'r') as f:
    lines = f.readlines()

brace_count = 0
bracket_count = 0
paren_count = 0

for i, line in enumerate(lines):
    b_open = line.count('{')
    b_close = line.count('}')
    sq_open = line.count('[')
    sq_close = line.count(']')
    p_open = line.count('(')
    p_close = line.count(')')
    
    brace_count += b_open - b_close
    bracket_count += sq_open - sq_close
    paren_count += p_open - p_close
    
    if brace_count < 0:
        print(f"Brace Underflow at line {i+1}: {line.strip()}")
        break
    if bracket_count < 0:
        print(f"Bracket Underflow at line {i+1}: {line.strip()}")
        break
    if paren_count < 0:
        print(f"Paren Underflow at line {i+1}: {line.strip()}")
        # break

print(f"Final Counts - Brace: {brace_count}, Bracket: {bracket_count}, Paren: {paren_count}")
if brace_count == 0 and bracket_count == 0 and paren_count == 0:
    print("Balanced!")
else:
    print("Imbalanced!")
